src/cheri_insts.sail

```diff
@@ -882,99 +749,63 @@ union clause ast = CBuildCap : (regidx, regidx, regidx)
+  let cd3 = setCapPerms(cd2, requestedPerms);
+  let cd4 = setCapFlags(cd3, requestedFlags);
+  let cd5 = if requestedSentry then sealCap(cd4, to_bits(cap_otype_width, otype_sentry)) else cd4;
+  let derivable = cd5 == requestedCap; /* True iff requestedCap has bounds not exceeding address space and no reserved bits set */
+  assert(not(derivable) | exact, "CBuildCap: setCapBounds was not exact"); /* If requestedCap was a derivable encoding then setBounds should be exact */
+  let cd6 = if subset & derivable then cd5 else clearTag(requestedCap);
+
+  C(cd) = cd6;
+  RETIRE_SUCCESS
 }
 
 union clause ast = CCopyType : (regidx, regidx, regidx)
 /*!
- * If the **otype** field of capability register *cs2* is not a reserved value
- * then capability register *cd* is replaced with the contents of capability
- * register *cs1* with the **address** set to *cs2*.**otype**. Otherwise
- * capability register *cd* is replaced with the **NULL** capability with its
- * **address** set to *cs2*.**otype**.
- *
- * ## Exceptions
- *
- * An exception is raised if:
- *   - *cs1*.**tag** is not set.
- *   - *cs1* is sealed.
- *   - *cs2*.**otype** $\lt$ *cs1*.**base**.
- *   - *cs2*.**otype** $\ge$ *cs1*.**top**.
+ * Capability register *cd* is replaced with the contents of capability
+ * register *cs1* with the **address** set to *cs2*.**otype** and the
+ * **tag** field cleared if *cs2* has a reserved **otype**.
  *
  * ## Notes
  *
- * - Reserved otypes always get turned into derived-from-**NULL** capabilities,
- *   regardless of the authority given in *cs1*, as, at the moment, all
- *   reserved otypes are constructed using ambiently-available actions.
- *   [CCSeal] knows how to work with these.
+ * - Reserved otypes always result in untagged capabilities, as, at the
+ *   moment, all reserved otypes are constructed using ambiently-available
+ *   actions. [CCSeal] knows how to work with these.
  */
 function clause execute (CCopyType(cd, cs1, cs2)) = {
   let cs1_val = C(cs1);
   let cs2_val = C(cs2);
-  let (cs1_base, cs1_top) = getCapBounds(cs1_val);
-  let cs2_otype = unsigned(cs2_val.otype);
-  if not (cs1_val.tag) then {
-    handle_cheri_reg_exception(CapEx_TagViolation, cs1);
-    RETIRE_FAIL
-  } else if isCapSealed(cs1_val) then {
-    handle_cheri_reg_exception(CapEx_SealViolation, cs1);
-    RETIRE_FAIL
-  } else if hasReservedOType(cs2_val) then {
-    C(cd) = int_to_cap(EXTS(cs2_val.otype));
-    RETIRE_SUCCESS
-  } else if cs2_otype < cs1_base then {
-    handle_cheri_reg_exception(CapEx_LengthViolation, cs1);
-    RETIRE_FAIL
-  } else if cs2_otype >= cs1_top then {
-    handle_cheri_reg_exception(CapEx_LengthViolation, cs1);
-    RETIRE_FAIL
-  } else {
-    let (success, cap) = setCapAddr(cs1_val, to_bits(cap_addr_width, cs2_otype));
-    assert(success, "CopyType: offset is in bounds so should be representable");
-    C(cd) = cap;
-    RETIRE_SUCCESS
-  }
+
+  let authCap = clearTagIfSealed(cs1_val);
```

Comment:
As with CBuildCap, not an authority

Would also move to next to the setCapAddr